### PR TITLE
Fix sorting on instrument column for generic investigation table

### LINKED ISSUE #767: Sorting on Instrument column on generic investigation table view not working
**Description**:
When trying to sort by instrument on generic investigation table, a 400 comes back from the API, thanks @sam-glendenning for the screenshots ;)
image

This is because of an incorrect order filter input, it should be `investigationInstruments.instrument.name`:
image

Having done a quick search of the codebase, it looks as if this is the only instance of it happening so doesn't need fixing elsewhere.

**Acceptance criteria**:
- [ ] On the generic investigation table, sorting by investigation should work, resulting in a 200 from the API request

### PR DESCRIPTION
**Pull request by MRichards99**
## Description
Tiny PR to fix #767. This means sorting on that column will occur on a one-to-many relationship, **Python ICAT 0.19.0 is required**, ideally with the changes from DataGateway API issue #238 (the corresponding PR for that brings both. If you don't have that, you'll get a 400 from the API request saying you cannot sort on a one to many relationship.

Screenshot below shows this works - both ascending and descending requests now return 200:
image

The second commit in this PR actually makes data present in that column, as proven when pointing at ISIS dev:
image

## Testing instructions
Discussed between Sam and I over Slack

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
connect to #767

### CODE REVIEW

File: packages/datagateway-dataview/src/views/table/investigationTable.component.tsx [hunk 1]
```diff
@@ -246,7 +246,16 @@ const InvestigationTable = (
         {
           icon: <AssessmentIcon />,
           label: t('investigations.instrument'),
-          dataKey: 'instrument.name',
+          dataKey: 'investigationInstruments.instrument.name',
+          cellContentRenderer: (cellProps: TableCellProps) => {
+            const investigationData = cellProps.rowData as Investigation;
+            if (investigationData?.investigationInstruments?.[0]?.instrument) {
+              return investigationData.investigationInstruments[0].instrument
```
Comment (sam-glendenning) [inline packages/datagateway-dataview/src/views/table/investigationTable.component.tsx:253]: I think if you wanna get this line covered, you'll have to change the test data from this:
```
state.dgcommon.data = [
    {
        ....
        instrument: {
            name: 'LARMOR',
        },
    }
]
```
to something like this (I borrowed it from `investigations.text.tsx`):
```
state.dgcommon.data = [
    {
        ....
        investigationInstruments: [
            {
                id: 3,
                instrument: {
                    id: 4,
                    name: 'LARMOR',
                },
            }
        ],
    }
]
```

**Review by sam-glendenning — CHANGES_REQUESTED** [review]

**Comment by MRichards99** [comment]
CI failures seem to be due to a timeout on download and a cypress issue (the deconstruct packet thing) on search. Will re-run them and see if it fixes itself

File: packages/datagateway-dataview/src/views/table/investigationTable.component.tsx (hunk 1 above)
Comment (MRichards99) [reply packages/datagateway-dataview/src/views/table/investigationTable.component.tsx:253]: I definitely would've forgotten to sort out the snapshot if it wasn't for the reminder :)

**Comment by MRichards99** [comment]
deconstruct packet errors on download and search this time, hmm

**Comment by MRichards99** [comment]
Think I'll leave this for the weekend and rerun on Monday morning to see if the errors have cleared

**Comment by MRichards99** [comment]
Seems as if the Cypress errors from Friday afternoon have fixed themselves this morning - ready for review :)

**Review by sam-glendenning — APPROVED** [review]

**Pull request merged**